cterasdk/edge/services.py

```diff
@@ -122,6 +122,10 @@ def _connect_to_services(self, param, ctera_license):
         task = self._attach(param)
         try:
             TaskManager.wait(self._gateway, task)
+            track(self._gateway, '/status/services/CTERAPortal/connectionState', [enum.ServicesConnectionState.Connected], [],
+                  [enum.ServicesConnectionState.ResolvingServers, enum.ServicesConnectionState.Connecting,
+                   enum.ServicesConnectionState.Attaching, enum.ServicesConnectionState.Authenticating],
+                  [enum.ServicesConnectionState.Disconnected], 5, 1)
```

Comment:
Maybe we should wait longer? Or let the user decide?

Reply:
Let's wait a bit longer. Changed to 20 retries. Theoretically, when the "Task" completes, the device should be connected but unfortunately the `connectionState` attribute is updated later